subsys/usb/device_next/Kconfig

```diff
@@ -16,6 +16,24 @@ module = USBD
 module-str = usbd
 source "subsys/logging/Kconfig.template.log_config"
 
+choice USBD_MAX_SPEED_CHOICE
+	prompt "Max supported connection speed"
+	default USBD_MAX_SPEED_FULL if UDC_IT82XX2 || UDC_KINETIS || UDC_NRF || UDC_STM32
+	default USBD_MAX_SPEED_HIGH
+
+config USBD_MAX_SPEED_HIGH
+	bool "High-Speed"
+
+config USBD_MAX_SPEED_FULL
+	bool "Full-Speed"
+
+endchoice
+
+config USBD_MAX_SPEED
+	int
+	default 0 if USBD_MAX_SPEED_FULL
+	default 1 if USBD_MAX_SPEED_HIGH
```

Comment:
How come this indirection is needed? Why not simple do `if (IS_ENABLED(CONFIG_USBD_MAX_SPEED_HIGH)` in the C code?

Reply:
The choice here makes it possible to add Super-Speed support later on without much hassle. `IS_ENABLED(CONFIG_USBD_MAX_SPEED_HIGH)` would make things harder IMHO.

Reply:
That makes sense, thanks. I'll take a closer look later today.